src/Http/HttpClientInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.
+ * Licensed under the MIT License.  See License in the project root
+ * for license information.
+ */
+
+namespace Microsoft\Graph\Http;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+
+/**
+ * Interface HttpClientInterface
+ *
+ * Provides methods for making synchronous and asynchronous requests on an HTTP client
+ *
+ * @package Microsoft\Graph\Http
+ * @copyright 2021 Microsoft Corporation
+ * @license https://opensource.org/licenses/MIT MIT License
+ * @link https://developer.microsoft.com/graph
+ */
+interface HttpClientInterface extends HttpClient, HttpAsyncClient
```

Comment:
I am not very sure about this but may be worth checking if empty interfaces are a clean 
https://stackoverflow.com/questions/7552677/are-empty-interfaces-code-smell

Reply:
Taking a look at this..

Reply:
*I think the StackOverflow discusses a different problem from the one here*

The StackOverflow discussion shared is around creating an **empty interface** (no functionality defined) that two different objects can "implement" so that the two objects can be passed to/returned interchangeably by one function.

In this use-case, we are **extending** two interfaces. By extending we are silently defining that the functionality defined in the two parent interfaces must be implemented by a class that uses our interface.

*Context on the problem we're trying to solve*

PHP has these two widely agreed upon interfaces defining entry points to send synchronous and asynchronous requests via a HTTP client. 
The problem is that we need a way to guarantee that a customer is passing an HTTP client implementation that can make both synchronous and asynchronous requests.
This seemed like the best approach in my view to enforce this.

I'm open to other ideas on how this can be achieved.